google/cloud/pubsub_v1/subscriber/_protocol/streaming_pull_manager.py

```diff
@@ -733,11 +863,20 @@ def _on_response(self, response: gapic_types.StreamingPullResponse) -> None:
             self._on_hold_bytes,
         )
 
+        with self._exactly_once_enabled_lock:
+            if response.subscription_properties.exactly_once_delivery_enabled != \
+               self._exactly_once_enabled:
+                 self._exactly_once_enabled = response.subscription_properties.exactly_once_delivery_enabled
+                 # Update ack_deadline, whose minimum depends on self._exactly_once_enabled
+                 # This method acquires the self._ack_deadline_lock lock.
+                 self._obtain_ack_deadline(maybe_update=True)
```

Comment:
Mahesh, look here for the logic to update the min ack deadline when we see `subscription_properties.exactly_once_delivery_enabled` has changed. I also have a test for this.